echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/HipchatNotificationAgent.groovy

```diff
@@ -76,17 +76,22 @@ class HipchatNotificationAgent extends AbstractEventNotificationAgent {
         message = """Stage ${event.content?.context?.stageDetails.name} for """
       }
 
+      String link = "${spinnakerUrl}/#/applications/${application}/${config.type == 'stage' ? 'executions/details' : config.link }/${event.content?.execution?.id}"
+
       message +=
-        """${WordUtils.capitalize(application)}'s <a href="${
-          spinnakerUrl
-        }/#/applications/${application}/${
-          config.type == 'stage' ? 'executions/details' : config.link
-        }/${event.content?.execution?.id}">${
+        """${WordUtils.capitalize(application)}'s <a href="${link}">${
           event.content?.execution?.name ?: event.content?.execution?.description
         }</a> ${buildInfo} ${config.type == 'task' ? 'task' : 'pipeline'} ${status == 'starting' ? 'is' : 'has'} ${
           status == 'complete' ? 'completed successfully' : status
         }"""
 
+      String customMessage = event.content?.context?.customMessage
+      if (customMessage && event.content?.execution?.id) {
+        message = customMessage
+          .replace("{{executionId}}", (String) event.content.execution.id)
```

Comment:
replaceAll ?

Reply:
would think about allowing also build number

Reply:
I don't think we want `replaceAll`, since we aren't looking at the tokens as regex patterns (`replace` does replace all matches, just...confusing API).

re: build number - should we make it available as "{{buildNumber}}" or "{{buildInfo}}"?